src/arc_spice/variational_pipelines/RTC_single_component_pipeline.py

```diff
@@ -0,0 +1,148 @@
+from collections.abc import Callable
+from typing import Any
+
+import torch
+from transformers import pipeline
+
+from arc_spice.variational_pipelines.RTC_variational_pipeline import (
+    CustomTranslationPipeline,
+    RTCPipelineBase,
+)
+from arc_spice.variational_pipelines.utils import dropout_off, dropout_on, set_dropout
+
+
+class RTCSingleComponentPipeline(RTCPipelineBase):
```

Comment:
would this be clearer if split this into 3 separate classes? (possibly keeping shared functionality here as an abstract class)
i.e. TranslationVariationalPipeline, RecognitionVariationalPipeline, TopicVariationalPipeline

Reply:
Sure, is this as subclasses of `RTCSingleComponentPipeline`?

Reply:
I've implemented this now